### PR TITLE
Adding a legacy Required properties resource mix models

### PR DESCRIPTION
**Pull request by allenjzhang**
Putting in a separate namespace avoid name conflict.

#131

### CODE REVIEW

File: packages/typespec-azure-resource-manager/lib/legacy/models.tsp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.OpenAPI;
+using Azure.ResourceManager.Foundations;
+using Azure.ResourceManager.Private;
+
+namespace Azure.ResourceManager.Legacy;
+
+//#region Standard Resource Operation Interfaces
```
Comment (timotheeguerin) [inline packages/typespec-azure-resource-manager/lib/legacy/models.tsp:9]: do we supprot region in tsp?

File: packages/typespec-azure-resource-manager/lib/legacy/models.tsp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * Concrete tracked resource types can be created by aliasing this type using a specific property type.
+ *
+ * See more details on [different Azure Resource Manager resource type here.](https://azure.github.io/typespec-azure/docs/howtos/ARM/resource-type)
+ * @template Properties A model containing the provider-specific properties for this resource
+ */
+@doc("Concrete tracked resource types can be created by aliasing this type using a specific property type.")
```
Comment (markcowl) [inline packages/typespec-azure-resource-manager/lib/legacy/models.tsp:16]: I think these changes should be made to the standard templates, and the older ones should be marked as Legacy and deprecated.

**Review by markcowl — CHANGES_REQUESTED** [review]

File: packages/typespec-azure-resource-manager/lib/legacy/models.tsp [hunk 3]
```diff
@@ -0,0 +1,59 @@
+using TypeSpec.Http;
```
Comment (markcowl) [inline packages/typespec-azure-resource-manager/lib/legacy/models.tsp:1]: For legacy, I think we are still leaving out scenarios where folks extend Resource or EntityResource.  I would prefer it if we just let them extend a common-types resource type, and added a new decorator they would need to decorate the type with that is the equivalent of `@armResourceInternal` and `@includeInapplicableMetadataInPayload`, and relax the rules around extension to include this case.

File: packages/typespec-azure-resource-manager/lib/legacy/models.tsp (hunk 1 above)
Comment (markcowl) [reply packages/typespec-azure-resource-manager/lib/legacy/models.tsp:9]: remove

File: packages/typespec-azure-resource-manager/lib/models.tsp [hunk 4]
```diff
@@ -40,7 +40,7 @@ model ResourceNameParameter<
 @armResourceInternal(Properties)
 @includeInapplicableMetadataInPayload(false)
 model TrackedResource<Properties extends {}> extends Foundations.TrackedResource {
-  @doc("The resource-specific properties for this resource.")
+  @doc("The resource-specific optional properties for this resource.")
```
Comment (markcowl) [inline packages/typespec-azure-resource-manager/lib/models.tsp:43]: I don't think these doc updates are necessary.  The typespec and the swagger already say it is optional, adding the documentation makes swagger changes without adding any information.

**Review by markcowl — CHANGES_REQUESTED** [review]
I think we should consider a different approach, since this leaves out many important brownfield scenarios.  Also, a couple of inline nits

File: packages/typespec-azure-resource-manager/lib/legacy/models.tsp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * Concrete tracked resource types can be created by aliasing this type using a specific property type.
+ *
+ * See more details on [different Azure Resource Manager resource type here.](https://azure.github.io/typespec-azure/docs/howtos/ARM/resource-type)
+ * @template Properties A model containing the provider-specific properties for this resource
+ */
+@doc("Concrete tracked resource types can be created by aliasing this type using a specific property type.")
+@armResourceInternal(Properties)
+@includeInapplicableMetadataInPayload(false)
+model TrackedResourceWithRequiredProperties<Properties extends {}>
```
[the remark this answers is not in the document]
Comment (allenjzhang) [reply packages/typespec-azure-resource-manager/lib/legacy/models.tsp:19]: I feel the namespace being legacy which has good separation. It is targeted for brownfield. Unless ARM has disallowed required properties (currently they don't) we probably don't want to put deprecate.

**Comment by allenjzhang** [comment]
This PR is ON HOLD until the discussion of @visibility design issues..

**Pull request closed without merging**